src/Servers/Kestrel/Core/src/Internal/Http/ForChunkedEncodingMessageBody.cs

```diff
@@ -0,0 +1,670 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http
+{
+    /// <summary>
+    ///   http://tools.ietf.org/html/rfc2616#section-3.6.1
+    /// </summary>
+    public class ForChunkedEncoding : Http1MessageBody
```

Comment:
This class has very little changed from the original implementation. It now contains the RequestBodyPipe rather than MessageBody and continues to call PumpAsync().